File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/bus/publish/RefreshEventPublish.java
package pers.acp.admin.oauth.bus.publish;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.cloud.bus.BusProperties;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;
import pers.acp.admin.oauth.bus.event.RefreshApplicationEvent;
import pers.acp.admin.oauth.bus.event.RefreshRuntimeEvent;

/**
 * @author zhang by 19/03/2019
 * @since JDK 11
 */
@Component
public class RefreshEventPublish {

    private final ApplicationContext applicationContext;

    private final BusProperties busProperties;

    @Autowired
    public RefreshEventPublish(ApplicationContext applicationContext, BusProperties busProperties) {
        this.applicationContext = applicationContext;
        this.busProperties = busProperties;
    }

    public void doNotifyUpdateApp() {
        RefreshApplicationEvent applicationEvent = new RefreshApplicationEvent(this, busProperties.getId(), null, "refresh client");
        applicationContext.publishEvent(applicationEvent);
    }

    public void doNotifyUpdateRuntime() {
        RefreshRuntimeEvent runtimeEvent = new RefreshRuntimeEvent(this, busProperties.getId(), null, "refresh runtime");
        applicationContext.publishEvent(runtimeEvent);
    }

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/bus/publish/RefreshEventPublish.kt
package pers.acp.admin.oauth.bus.publish

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.cloud.bus.BusProperties
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Component
import pers.acp.admin.common.event.ReloadDataBusEvent
import pers.acp.admin.oauth.constant.BusEventMessage

/**
 * @author zhang by 19/03/2019
 * @since JDK 11
 */
@Component
class RefreshEventPublish @Autowired
constructor(private val applicationContext: ApplicationContext, private val busProperties: BusProperties) {

    fun doNotifyUpdateApp() {
        val source = this
        GlobalScope.launch(Dispatchers.IO) {
            applicationContext.publishEvent(ReloadDataBusEvent(busProperties.id, null, BusEventMessage.refreshApplication, source))
        }
    }

    fun doNotifyUpdateRuntime() {
        val source = this
        GlobalScope.launch(Dispatchers.IO) {
            applicationContext.publishEvent(ReloadDataBusEvent(busProperties.id, null, BusEventMessage.refreshRuntime, source))
        }
    }

}
